Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.h

```diff
@@ -236,5 +318,130 @@ namespace Terrain
         AZStd::map<AZ::EntityId, TerrainAreaData, TerrainLayerPriorityComparator> m_registeredAreas;
 
         mutable TerrainRaycastContext m_terrainRaycastContext;
+
+        AZ::JobManager* m_terrainJobManager = nullptr;
     };
+
+    template<typename SynchronousFunctionType, typename VectorType>
+    inline AZStd::shared_ptr<AzFramework::Terrain::TerrainDataRequests::TerrainJobContext> TerrainSystem::ProcessFromListAsync(
+        SynchronousFunctionType synchronousFunction,
+        const AZStd::span<VectorType>& inPositions,
+        AzFramework::Terrain::SurfacePointListFillCallback perPositionCallback,
+        Sampler sampleFilter,
+        AZStd::shared_ptr<ProcessAsyncParams> params) const
+    {
+        // Determine the number of jobs to split the work into based on:
+        // 1. The number of available worker threads.
+        // 2. The desired number of jobs as passed in.
+        // 3. The number of positions being processed.
+        const int32_t numWorkerThreads = m_terrainJobManager->GetNumWorkerThreads();
+        const int32_t numJobsDesired = params ? params->m_desiredNumberOfJobs : ProcessAsyncParams::NumJobsDefault;
+        const int32_t numJobsMax = (numJobsDesired > 0) ? AZStd::min(numWorkerThreads, numJobsDesired) : numWorkerThreads;
+        const int32_t numPositionsToProcess = static_cast<int32_t>(inPositions.size());
+        const int32_t numJobs = AZStd::min(numJobsMax, numPositionsToProcess);
+        if (numJobs <= 0)
+        {
+            AZ_Warning("TerrainSystem", false, "No positions to process.");
+            return nullptr;
+        }
+
+        // Create a terrain job context to return so the jobs can be cancelled.
+        AZStd::shared_ptr<TerrainJobContext> jobContext = AZStd::make_shared<TerrainJobContext>(*m_terrainJobManager);
+
```

Comment:
Turns out that doesn't work at all because the JobCancelGroup stores raw pointers to child job cancel groups, so if the child job cancel group gets destroyed we're left with dangling pointers.

Reply:
I re-introduced global cancellation of all terrain jobs on deactivation of the TerrainSystem using a different way (not using JobCancelGroups at all anymore).